meson.build

```diff
@@ -68,13 +68,12 @@ if brotli_dep.found()
 endif
 
 if get_option('webready')
-  curl_dep = dependency('libcurl', disabler: true, required: get_option('curl'))
+  curl_dep = dependency('libcurl', required: get_option('curl'))
+  web_dep = declare_dependency(dependencies: [curl_dep])
+  deps += web_dep
 else
-  curl_dep = dependency('', disabler: true, required: false)
-endif
-
-if curl_dep.found()
-  deps += curl_dep
+  web_dep = dependency('', disabler: true, required: false)
```

Comment:
Is disabler correct here?

Reply:
@neheb

It is, though apparently I could just directly assign a `disabler()` to `web_dep`, as that's effectively what this will do as a [default not-found dependency](https://mesonbuild.com/howtox.html#get-a-default-notfound-dependency).

The disabler returned here is what automatically excludes the `conntest` sample from the build, when the `webready` option is `false`.

(`conntest` now uses `web_dep` as a dependency. Previously it used `curl_dep`, which had been set as the disabler. But `conntest` doesn't actually require curl, it can build without it as long as `webready` is enabled in the build. If webready isn't enabled, though, `conntest` has to be _dis_-abled.)

Reply:
OK